tests/e2e/datadog/logs/mod.rs

```diff
@@ -0,0 +1,77 @@
+use serde_json::Value;
+use tracing::info;
+
+use vector::test_util::trace_init;
+
+use super::*;
+
+const LOGS_ENDPOINT: &str = "/api/v2/logs";
+
+fn expected_log_events() -> usize {
+    std::env::var("EXPECTED_LOG_EVENTS")
+        .map(|n_expected| {
+            n_expected
+                .parse::<usize>()
+                .expect("EXPECTED_LOG_EVENTS should be an unsigned integer.")
+        })
+        .unwrap_or(1000)
+}
+
+// Asserts that each log event has the hostname and timestamp fields, and
+// Removes them from the log so that comparison can more easily be made.
+// @return the number of log entries in the payload.
+fn assert_timestamp_hostname(payloads: &mut [Value]) -> usize {
+    let mut n_log_events = 0;
+
+    payloads.iter_mut().for_each(|payload_array| {
+        payload_array
+            .as_array_mut()
+            .expect("should be array")
+            .iter_mut()
+            .for_each(|log_val| {
+                n_log_events += 1;
+
+                let log = log_val
+                    .as_object_mut()
+                    .expect("log entries should be objects");
+
+                assert!(log.remove("timestamp").is_some());
```

Comment:
Is this needed because the original log lines generated by flog do not have timestamps, so a timestamp is assigned in the agent?

Reply:
In a nutshell yes. There is a timestamp in the flog generated logs (it supports a few different log formats) but AFAICT none meet the requirements for automatic parsing in the Agent. I looked briefly at adding a custom parser but it appears that is done in the DD ui and not in the agent.

In the metrics tests (other pr), we ignore the timestamp entirely because we have separate emitters. So while we might be able to get it sorted in the logs tests here to not have to remove this, I think the value isn't super high.

But I will add a comment here explaining.